forms/HeaderField.php

```diff
@@ -8,52 +9,80 @@
  * @subpackage fields-dataless
  */
 class HeaderField extends DatalessField {
-	
 	/**
-	 * @var int $headingLevel The level of the <h1> to <h6> HTML tag. Default: 2
+	 * The level of the <h1> to <h6> HTML tag.
+	 *
+	 * @var int
 	 */
 	protected $headingLevel = 2;
-	
+
+	/**
+	 * @param string $name
+	 * @param null|string $title
+	 * @param int $headingLevel
+	 */
 	public function __construct($name, $title = null, $headingLevel = 2) {
-		// legacy handling for old parameters: $title, $heading, ...
-		// instead of new handling: $name, $title, $heading, ...
+		// legacy handling:
+		// $title, $headingLevel...
 		$args = func_get_args();
+
 		if(!isset($args[1]) || is_numeric($args[1])) {
-			$title = (isset($args[0])) ? $args[0] : null;
-			// Use "HeaderField(title)" as the default field name for a HeaderField; if it's just set to title then we
-			// risk causing accidental duplicate-field creation.
-			// this means i18nized fields won't be easily accessible through fieldByName()
+			if(isset($args[0])) {
+				$title = $args[0];
+			}
+
+			// Prefix name to avoid collisions.
 			$name = 'HeaderField' . $title;
-			$headingLevel = (isset($args[1])) ? $args[1] : null;
-			$form = (isset($args[3])) ? $args[3] : null;
-		} 
-		
-		if($headingLevel) $this->headingLevel = $headingLevel;
-		
+
+			if(isset($args[1])) {
+				$headingLevel = $args[1];
+			}
+
+			if(isset($args[3])) {
+				$this->setForm($args[3]);
```

Comment:
The old code ignores $form and passing in $form into the field contstructor has been deprecated since 3.x.0 (can't recall exactly). Same argument as https://github.com/silverstripe/silverstripe-framework/pull/4128/files... better to remove the unused variable than try to call setForm where it wasn't called before.